jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -1523,12 +1523,17 @@ public void succeeded()
                 return;
             }
 
+            // as this is not an upgrade, we can shutdown the output if we know we are not persistent
+            if (_sendCallback._shutdownOut)
+                getEndPoint().shutdownOutput();
+
             _httpChannel.recycle();
 
+
+            // If we are still expecting 100 Continue and no content was read, then
+            // close the parser so that below it seeks EOF, not the next request.
```

Comment:
The comment was more correct before, as we are not expecting 100 Continue, we test whether we have sent it.